vllm/model_executor/layers/quantization/compressed_tensors/scaled_mm_triton.py

```diff
@@ -0,0 +1,184 @@
+from typing import Optional, Type
+
+import torch
+import triton
+import triton.language as tl
+
+
+def has_good_tensor_strides(x: torch.Tensor):
+    strides = x.stride()
+    sizes = x.shape
+    is_not_transpose = strides[0] == 1 and (strides[1] >= max(1, sizes[0]))
+    is_transpose = strides[1] == 1 and (strides[0] >= max(1, sizes[1]))
+    return is_transpose or is_not_transpose
```

Comment:
Could you change the function name to be more descriptive? Rather than `good`, it would be better for the name to convey why the tensor strides are valid. I suggest something like `is_weak_contiguous` or `is_contiguous_or_transpose_contiguous`

Reply:
Done